AppCenterDataStorage/AppCenterDataStorage/Internal/LocalStore/MSDBDocumentStore.m

```diff
@@ -53,65 +55,71 @@ - (instancetype)init {
   int result = [self.dbStorage executeNonSelectionQuery:insertQuery];
   if (result != SQLITE_OK) {
     MSLogError([MSDataStore logTag], @"Unable to update or replace stored document, SQLite error code: %ld", (long)result);
   }
   return result == SQLITE_OK;
 }
 
-- (BOOL)deleteWithPartition:(NSString *)partition documentId:(NSString *)documentId {
-  NSString *tableName = [MSDBDocumentStore tableNameForPartition:partition];
+- (BOOL)deleteWithToken:(MSTokenResult *)token documentId:(NSString *)documentId {
+  NSString *tableName = [MSDBDocumentStore tableNameForPartition:token.partition];
   NSString *deleteQuery = [NSString stringWithFormat:@"DELETE FROM \"%@\" WHERE \"%@\" = '%@' AND \"%@\" = '%@'", tableName,
-                                                     kMSPartitionColumnName, partition, kMSDocumentIdColumnName, documentId];
+                                                     kMSPartitionColumnName, token.partition, kMSDocumentIdColumnName, documentId];
   int result = [self.dbStorage executeNonSelectionQuery:deleteQuery];
   if (result != SQLITE_OK) {
     MSLogError([MSDataStore logTag], @"Unable to delete stored document, SQLite error code: %ld", (long)result);
   }
   return result == SQLITE_OK;
 }
 
-- (BOOL)createUserStorageWithAccountId:(NSString *)accountId {
-
-  // Create table based on the schema.
-  return [self.dbStorage createTable:[NSString stringWithFormat:kMSUserDocumentTableNameFormat, accountId]
-                       columnsSchema:[MSDBDocumentStore columnsSchema]
-             uniqueColumnsConstraint:@[ kMSPartitionColumnName, kMSDocumentIdColumnName ]];
-}
-
 - (BOOL)deleteUserStorageWithAccountId:(NSString *)accountId {
   NSString *tableName = [NSString stringWithFormat:kMSUserDocumentTableNameFormat, accountId];
   return [self.dbStorage dropTable:tableName];
 }
 
-- (MSDocumentWrapper *)readWithPartition:(NSString *)partition
-                              documentId:(NSString *)documentId
-                            documentType:(Class)documentType
-                             readOptions:(MSReadOptions *)__unused readOptions {
-  NSString *selectionQuery =
-      [NSString stringWithFormat:@"SELECT * FROM \"%@\" WHERE \"%@\" = \"%@\" AND \"%@\" = \"%@\"", kMSAppDocumentTableName,
-                                 kMSPartitionColumnName, partition, kMSDocumentIdColumnName, documentId];
+- (MSDocumentWrapper *)readWithToken:(MSTokenResult *)token documentId:(NSString *)documentId documentType:(Class)documentType {
+  NSString *tableName = [MSDBDocumentStore tableNameForPartition:token.partition];
+  NSString *selectionQuery = [NSString stringWithFormat:@"SELECT * FROM \"%@\" WHERE \"%@\" = \"%@\" AND \"%@\" = \"%@\"", tableName,
+                                                        kMSPartitionColumnName, token.partition, kMSDocumentIdColumnName, documentId];
   NSArray *result = [self.dbStorage executeSelectionQuery:selectionQuery];
```

Comment:
I wonder the query should be dispatch in a separate queue in order to no block current queue.
@achocron, @YiMysty added that to their PR here:
https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1489

Reply:
@MehrdadMzfr We are putting it in a queue from the data store level, not the local storage level, so this should be fine :)